prov/shm/src/smr_atomic.c

```diff
@@ -196,12 +192,14 @@ static ssize_t smr_generic_atomic(struct smr_ep *ep,
 
 	domain = container_of(ep->util_ep.domain, struct smr_domain, util_domain);
 
-	peer_id = (int) addr;
-	ret = smr_verify_peer(ep, peer_id);
+	id = (int) addr;
+	peer_id = smr_peer_addr(ep->region)[id].addr;
+
+	ret = smr_verify_peer(ep, id);
 	if(ret)
 		return ret;
 
-	peer_smr = smr_peer_region(ep->region, peer_id);
+	peer_smr = smr_peer_region(ep->region, id);
```

Comment:
Is this bug fix needed in v1.9.x?

Reply:
no, this is just refactoring

Reply:
I don't know if this is the bug fix.  I'm referring to the bug fix in the commit message, which should be pulled out into its own commit.

Reply:
Gotcha, will pull that bug fix out and include in the new PR